Sources/Commands/SwiftPackageTool.swift

```diff
@@ -966,3 +1051,207 @@ fileprivate func logPackageChanges(changes: [(PackageReference, Workspace.Packag
+        }
+        return
+    } else {
+        // These are only needed in the event that --type was not used when creating a package
+        // otherwise if --type is used packageType will not be nill
+        let defualtType = mode == .initialize ? InitPackage.PackageType.library : InitPackage.PackageType.executable
+        packageTemplate = try InitPackage.PackageTemplate(template: getSwiftPMDefaultTemplate(type: packageType ?? defualtType))
+    }
+
+    let initPackage = try InitPackage(name: name, destinationPath: destinationPath, packageTemplate: packageTemplate)
+    initPackage.progressReporter = { message in
+        print(message)
+    }
+    try initPackage.writePackageStructure()
+}
+
+fileprivate func copyTemplate(fileSystem: FileSystem, sourcePath: AbsolutePath, destinationPath: AbsolutePath, name: String) throws {
+    // Recursively copy the template package
+    // Currently only replaces the string literal "$NAME"
+    if fileSystem.isDirectory(sourcePath) {
+        if let fileName = sourcePath.pathString.split(separator: "/").last {
+            if !fileSystem.exists(destinationPath.appending(component: String(fileName))) {
+                try fileSystem.createDirectory(destinationPath.appending(component: String(fileName)))
+            }
+            
+            try fileSystem.getDirectoryContents(sourcePath).forEach {
+                try copyTemplate(fileSystem: fileSystem,
+                                 sourcePath: sourcePath.appending(component: $0),
+                                 destinationPath: destinationPath.appending(components: String(fileName)),
+                                 name: name)
+            }
+        }
+    } else {
+        let fileContents = try fileSystem.readFileContents(sourcePath)
+        
+        if let validDescription = fileContents.validDescription {
+            var renamed = validDescription.replacingOccurrences(of: "___NAME___", with: name)
+            renamed = renamed.replacingOccurrences(of: "___NAME_AS_C99___", with: name.spm_mangledToC99ExtendedIdentifier())
+            
+            if let fileName = sourcePath.pathString.split(separator: "/").last {
+                if !fileSystem.exists(destinationPath.appending(component: String(fileName))) {
+                    try fileSystem.writeFileContents(destinationPath.appending(component: String(fileName))) { $0 <<< renamed }
+                }
+            }
+        } else {
+            // This else takes care of things such as images
+            if let fileName = sourcePath.pathString.split(separator: "/").last {
+                if !fileSystem.exists(destinationPath.appending(component: String(fileName))) {
+                    try fileSystem.copy(from: sourcePath, to: destinationPath.appending(component: String(fileName)))
+                }
+            }
+        }
+    }
+}
+
+fileprivate enum MakePackageMode {
```

Comment:
this enum should be be in an extension to InitPackage rather than "free floating" to better encapsulate the logic